### PR TITLE
metaxy metadata copy CLI

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary by cubic
Add a metadata copy CLI and backend API to copy feature metadata between stores with snapshot-aware filtering. Preserves snapshot_id and feature_version, and supports fine-grained filters.

- **New Features**
  - New CLI: metaxy metadata copy with --from, --to, --feature (repeatable), --all-features, and --snapshot flags.
  - MetadataStore.copy_metadata: copies all or selected features; uses latest or explicit snapshot; supports global and per-feature filters via FilteredFeature; auto-opens the source store; returns copy stats.
  - Helper APIs: read_graph_snapshots and read_features for querying snapshots and feature versions; FilteredFeature exported from metaxy.metadata_store.
  - Tests: comprehensive coverage for snapshots, filters, mixed feature inputs, and edge cases.

### CODE REVIEW

**Comment by danielgafni** [comment]
* **#24** <a href="https://app.graphite.dev/github/pr/anam-org/metaxy/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23** <a href="https://app.graphite.dev/github/pr/anam-org/metaxy/23?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/anam-org/metaxy/23?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/metaxy/metadata_store/base.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
 """Abstract base class for metadata storage backends."""
 
+from __future__ import annotations
```
Comment (geoHeil) [inline src/metaxy/metadata_store/base.py]: same

File: tests/metadata_stores/test_copy_metadata.py [hunk 2]
```diff
@@ -0,0 +1,741 @@
+"""Tests for metadata store copy functionality."""
+
+from __future__ import annotations
```
Comment (geoHeil) [inline tests/metadata_stores/test_copy_metadata.py]: again

File: src/metaxy/cli/metadata.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
+"""Metadata management commands for Metaxy CLI."""
+
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (danielgafni) [reply src/metaxy/cli/metadata.py]: Yeah will clean them up

**Comment by danielgafni** [comment]
@cubic-dev-ai stop managing this repo
